### PR TITLE
supports customized kafka client id

### PR DESCRIPTION
**Pull request by newly12**
## Which problem is this PR solving?
#1506 

## Short description of the changes
supports customized kafka client id in ingester

### CODE REVIEW

File: plugin/storage/integration/kafka_test.go [hunk 1]
```diff
@@ -46,6 +46,7 @@ func (s *KafkaIntegrationTestSuite) initialize() error {
 	s.logger, _ = testutils.NewLogger()
 	const encoding = "json"
 	const groupID = "kafka-integration-test"
+	const ClientID = "kafka-integration-test"
```
Comment (vprithvi) [inline plugin/storage/integration/kafka_test.go]: nit: this can be private

Comment (newly12) [reply plugin/storage/integration/kafka_test.go]: Thanks. it's updated.

File: cmd/ingester/app/flags.go [hunk 2]
```diff
@@ -64,6 +68,7 @@ const (
 // Options stores the configuration options for the Ingester
 type Options struct {
 	kafkaConsumer.Configuration
+	ClientID         string
```
Comment (vprithvi) [inline cmd/ingester/app/flags.go]: I didn't notice this earlier - but we should move this to `kafkaConsumer.Configuration` which is where GroupID/Topic/etc are configured.

Comment (newly12) [reply cmd/ingester/app/flags.go]: yes it is redundant. it is removed.

**Comment by vprithvi** [comment]
Thanks for the contribution!

**Pull request merged**